### PR TITLE
Prediction example

### PR DESCRIPTION
**Pull request by UzairNoman**
I have added the prediction example to users can use to convert their source images to target using gen_ab model.

### CODE REVIEW

File: examples/predict.py
Comment (usert5432) [inline examples/predict.py:52]: Perhaps, it would be neater if one can define a new function, like `create_data_loader` and put this code block inside of that function? This change may help the reader to more easily navigate the logic of the code.

Comment (usert5432) [reply examples/predict.py:52]: Also, could I ask you please to limit the line size to 79 characters (as suggested by the PEP8 guidelines)?

File: examples/predict.py
Comment (usert5432) [inline examples/predict.py:49]: Since this is an example code, perhaps we can omit the explicit path, and just put a placeholder, like 
```python
model_path = "<path_to_saved_model>"
```
?

File: examples/predict.py
Comment (usert5432) [inline examples/predict.py:48]: Perhaps, this may be a good place to define additional constants used in the script, like `epoch` and `dataset_path`?

**Review by usert5432 — COMMENTED** [review]
Hi @UzairNoman. Thank you again for your submission! I think your prediction example will be a great addition to the `uvcgan` package, since our current prediction script `scripts/translate_images` is too abstract and inflexible. However, to ensure the code style consistency with the rest of the codebase, I would like to ask you please to make a few changes to your example. After that, I will take another pass through the PR (to fix some cosmetic issues) and we would be able to merge it.

**Comment by UzairNoman** [comment]
Hi, glad to contribute to the repo. Made changes to the PR. Let me know if any change is required 🚀

**Comment by UzairNoman** [comment]
Oh yes, CycleGANDataset it should be. Sounds reasonable! I have resolved those issues. Now receiving the code rating of 10/10 on pylint. Let me if anything else needs changing :)

**Comment by usert5432** [comment]
Thank you, @UzairNoman! Looks good. Now, let me quickly run this through my supervisor to make sure they are ok with the changes and we can merge this PR!

File: examples/predict.py
Comment (YHRen) [inline examples/predict.py:49]: pass the `batch_size` as an argument.

**Comment by UzairNoman** [comment]
Made the marked readability changes. Let me know if it looks better now.

**Comment by usert5432** [comment]
Hi @UzairNoman, Thank you for making another round of revisions!

I have looked over the recent changes. And, I have a quick suggestion, in case if you may find it helpful. As you may have noticed, there is a variable name clash between the global namespace and local namespaces of different functions. This is what `pylint` indicates as:
```
W0621: Redefining name 'NAME' from outer scope (line N) (redefined-outer-name)
```

You have resolved these `pylint` issues by giving the variables different names (e.g. `model_path` vs `model_pth`).

However, this may be a suboptimal solution, since it requires a lot of renaming. There is a one accepted approach to easily fix the problem -- people usually rename outer (global) variables in `UPPER_SNAKE_CASE` and keep local variables in `snake_case`. For example, the common way to rewrite your code will be

```python
def load_gen_ab(model_path, epoch):
    ...

def create_data_loader(data_path, batch_size):
    ...

if __name__ == '__main__':
    BATCH_SIZE = 32
    EPOCH = 200
    DEVICE = get_torch_device_smart()

    DATA_PATH = "<path_to_your_data>"
    MODEL_PATH = "<path_to_saved_model>"

    MODEL = load_gen_ab(MODEL_PATH, EPOCH)
    DL = create_data_loader(DATA_PATH, BATCH_SIZE)

    making_predictions(DL, MODEL)
```

As you can see, in this case, you do not have to come up with creative names (e.g. `model_path` vs `model_pt`), or pack parameters into a dictionary (e.g. `hyper_params`). Instead, you simply make the global variables UPPER_CASE.

**Review by YHRen — APPROVED** [review]
Thanks for the changes. Feel free to merge.

**Comment by UzairNoman** [comment]
Hi @usert5432, did you manage to look at it? Can we merge this PR?

**Comment by usert5432** [comment]
Hi @UzairNoman. The PR is good to be merged if you ask me. However, I am still waiting on my supervisor's approval. Let me bug them again...

**Comment by usert5432** [comment]
Hi @UzairNoman,

I have received a feedback from my supervisor. They suggest a few things:
1. They recommend you to put your copyright at the top of the example file with your name/email address. This will increase your visibility and help others to recognize your work. For example, you can add the following line near the top of the file:
```python
# Copyright (c) 2022 Your Name <Your@Email>
```
2. Similarly, since it is your work, you can choose any license to publish it under. We recommend you to use the BSD-2 license (bacause the rest of the code has it) and add another line near the top of the file, like:
```python
# BSD 2-Clause License
```
Again, you can choose any other license to publish your work under, but please let us know first to make sure that your license is compatible with ours.
3. Finally, the current `examples/` directory is a bit barren and users may get lost if they stumble into it. My supervisor suggests, that you create a `README.md` file in there (like `examples/README.md`) and move there the description of your example from the top-level readme file.

We, then, modify the top-level `README.md`ourselves and point the users to your example.

**Comment by UzairNoman** [comment]
Hi, made the three changes.

**Comment by usert5432** [comment]
Perfect, it all looks great, I am going to merge the PR!

Thank you very much @UzairNoman for taking your time to make a contribution and help us to improve the `uvcgan` package! We appreciate your work!

And thank you for following through our sluggish review process, I know how frustrating that can be.

**Pull request merged**
